recurring_contract/model/contract_group.py

```diff
@@ -82,22 +91,102 @@ def _get_groups_from_contract(self, cr, uid, ids, context=None):
                     _get_groups_from_contract, ['next_invoice_date',
                                                 'state'], 20),
             }),
+        'last_paid_invoice_date': fields.function(
+            _get_last_paid_invoice, type='date',
+            string=_('Last paid invoice date')
+        )
     }
 
     _defaults = {
         'ref': '/',
         'recurring_unit': 'month',
         'recurring_value': 1,
+        'advance_billing_months': 1,
     }
 
+    def _get_contract_ids(self, cr, uid, ids, context=None):
+        contract_ids = list()
+
+        for contract_group_id in self.browse(cr, uid, ids, context):
+            for contract in contract_group_id.contract_ids:
+                if contract.id not in contract_ids:

```

Comment:
Can the same contract be in several groups ? I don't think so...  I think this test is useless. As well the whole procedure could be reduced to :
```contract_ids.extend([contract.id for contract in contract_group.contract_ids])```